src/services/SlotService.ts

```diff
@@ -52,8 +54,32 @@ export default class SlotService implements IService<Slot> {
     return slot;
   }
 
-  updateById(): Promise<Slot> {
-    throw new Error("Method not implemented.");
+  /**
+   * Updates slot (Disable slots and deletes bookings linked to it)
+   * @param slotData
+   * @returns slot
+   */
+
+
+  public async updateById(
+    id: number,
+    slotData: SlotUpdateBodyValidator,
+  ): Promise<Slot> {
+    const repo = this.getRepository();
+    const slot = await this.getByKey(
+      "id", id,
+    );
+
+    const bookingService = new BookingService();
+
+    if (slot.isDisabled === false) {
```

Comment:
What happens if the slot update (on the `repo.save`) fails say for a database connection issue :) ?